### ISSUE TITLE
Please log into this app again to reconnect your Facebook account 

### ISSUE THREAD

**Issue by mogsten** [ParsePlatform/parse-server#796]
When logging in with Facebook, ParseFacebookUtilsV4 and Parse Server, I receive the message
"Please log into this app again to reconnect your Facebook account", and then the app crashes on the line that gets the facebook results as an NSDictionary, is there anything that could be causing an issue

Parse Server is on the latest version
facebookAppIds and fileKey is set correctly and working

**Comment by drew-gross** [comment]
Does this same code work when using Parse.com instead of Parse Server? If yes, can you show the request and response from your Parse Server?

**Comment by mogsten** [comment]
Thanks for your reply, 
Yes the code works correctly when using Parse.com, the error only occurs when I use Parse Server. How would I be able to get the request and response from the Parse Server?

**Comment by FridayDevGroup** [comment]
@mogsten What error are you getting from the console?

**Comment by mogsten** [comment]
Which Console?

iOS?
or another console

**Comment by FridayDevGroup** [comment]
iOS

**Comment by mogsten** [comment]
<img width="1680" alt="screen shot 2016-03-04 at 11 29 34 am" src="https://cloud.githubusercontent.com/assets/10417933/13514864/6bb86baa-e1fc-11e5-8885-9903df0130c6.png">

This is the issue that I am gettings

**Comment by FridayDevGroup** [comment]
@mogsten Can you try "print(error)" before the line that is crashing? (Before the 'let' statement).

**Comment by mogsten** [comment]
ERROR: Error Domain=com.facebook.sdk.core Code=8 "(null)" UserInfo={NSRecoveryAttempter=<_FBSDKLoginRecoveryAttempter: 0x7fb7eb201da0>, NSLocalizedRecoverySuggestion=Please log into this app again to reconnect your Facebook account., com.facebook.sdk:FBSDKGraphRequestErrorGraphErrorCode=190, com.facebook.sdk:FBSDKErrorDeveloperMessageKey=Error validating access token: The session has been invalidated because the user has changed the password., com.facebook.sdk:FBSDKGraphRequestErrorHTTPStatusCodeKey=400, com.facebook.sdk:FBSDKGraphRequestErrorCategoryKey=2, com.facebook.sdk:FBSDKGraphRequestErrorGraphErrorSubcode=460, com.facebook.sdk:FBSDKGraphRequestErrorParsedJSONResponseKey={
    body =     {
        error =         {
            code = 190;
            "error_subcode" = 460;
            "fbtrace_id" = C8G9qKdEqg8;
            message = "Error validating access token: The session has been invalidated because the user has changed the password.";
            type = OAuthException;
        };
    };
    code = 400;
}, NSLocalizedRecoveryOptions=(
    OK,
    Cancel
)}
fatal error: unexpectedly found nil while unwrapping an Optional value

**Comment by mogsten** [comment]
Has there been any resolution for this case?

**Comment by flovilmart** [comment]
@mogsten you're doing a force unpack (!) on your result that's why you crash

Your should do:
```
guard let result = result as? [NSObject:AnyObject] else {
  if let error  = error { print(error) }
  return 
}

// continue with the result
```

**Comment by mogsten** [comment]
User signed up and logged in through Facebook!
ERROR: Error Domain=com.facebook.sdk.core Code=8 "(null)" UserInfo={NSRecoveryAttempter=<_FBSDKLoginRecoveryAttempter: 0x15e68e1c0>, NSLocalizedRecoverySuggestion=Please log into this app again to reconnect your Facebook account., com.facebook.sdk:FBSDKGraphRequestErrorGraphErrorCode=190, com.facebook.sdk:FBSDKErrorDeveloperMessageKey=Error validating access token: Session does not match current stored session. This may be because the user changed the password since the time the session was created or Facebook has changed the session for security reasons., com.facebook.sdk:FBSDKGraphRequestErrorHTTPStatusCodeKey=400, com.facebook.sdk:FBSDKGraphRequestErrorCategoryKey=2, com.facebook.sdk:FBSDKGraphRequestErrorGraphErrorSubcode=460, com.facebook.sdk:FBSDKGraphRequestErrorParsedJSONResponseKey={
    body =     {
        error =         {
            code = 190;
            "error_subcode" = 460;
            "fbtrace_id" = EeL1NzTdqu0;
            message = "Error validating access token: Session does not match current stored session. This may be because the user changed the password since the time the session was created or Facebook has changed the session for security reasons.";
            type = OAuthException;
        };
    };
    code = 400;
}, NSLocalizedRecoveryOptions=(
    OK,
    Cancel
)}
Error Domain=com.facebook.sdk.core Code=8 "(null)" UserInfo={NSRecoveryAttempter=<_FBSDKLoginRecoveryAttempter: 0x15e68e1c0>, NSLocalizedRecoverySuggestion=Please log into this app again to reconnect your Facebook account., com.facebook.sdk:FBSDKGraphRequestErrorGraphErrorCode=190, com.facebook.sdk:FBSDKErrorDeveloperMessageKey=Error validating access token: Session does not match current stored session. This may be because the user changed the password since the time the session was created or Facebook has changed the session for security reasons., com.facebook.sdk:FBSDKGraphRequestErrorHTTPStatusCodeKey=400, com.facebook.sdk:FBSDKGraphRequestErrorCategoryKey=2, com.facebook.sdk:FBSDKGraphRequestErrorGraphErrorSubcode=460, com.facebook.sdk:FBSDKGraphRequestErrorParsedJSONResponseKey={
    body =     {
        error =         {
            code = 190;
            "error_subcode" = 460;
            "fbtrace_id" = EeL1NzTdqu0;
            message = "Error validating access token: Session does not match current stored session. This may be because the user changed the password since the time the session was created or Facebook has changed the session for security reasons.";
            type = OAuthException;
        };
    };
    code = 400;
}, NSLocalizedRecoveryOptions=(
    OK,
    Cancel
)}


This is what was returned, fixed the crashing but still having issues

**Comment by flovilmart** [comment]
This is an iOS Facebook SDK problem, try to reach out on stack overflow.

**Comment by Gegham** [comment]
I have the same problem, have you solved this problem on your side ? @mogsten

**Comment by umales** [comment]
Same problem... @Gegham  @mogsten

**Comment by roylo** [comment]
Same problem.... Does anyone solve this?

**Comment by FridaySG** [comment]
Try updating to the latest version of the Facebook SDK, in SDK changelog they mentioned this issue. This is not a Parse problem.

**Comment by sdalai** [comment]
I just got same problem. I upgrade to latest SDK. Then got this problem. Has anybody resolved. Does my  privately hosted parse server need to have secure connection?

**Comment by sdalai** [comment]
I just logged an issue
Here is my scenario
I have migrated to linux based Private parse environment in Rack space.
I use PF Login for user login using Facebook.

Here is Latest versions
Parse Server 2.2.6
FBSDKCoreKit 4.11.0
ParseUI (1.2.0)
ParseFacebookUtilsV4 (1.11.1)
IOS 8 and above

Everything have been working fine. Then I changed my Facebook password. When I am logging into app using Face book alert is coming up with message

"Please log into this app again to reconnect to your Facebook account."

I am logging out and login in again to get same message.

However other users who haven't changed password are able to login as usual.

To further check
I pointed my app to parse.com. It is working fine .

**Comment by FilipZawada** [comment]
@flovilmart Requesting another token doesn't help here (as well as displaying a login screen).

Anyway, in old Parse dashboard we were asked to provide `facebook app id` AND `facebook secret`. In open-source parse however, we don't need to provide facebook secret. Why is that, what is the difference between the old and new parse and the secret thing? Can it cause similar problems as described above?

**Comment by flovilmart** [comment]
@FilipZawada The secret is only used in particular cases, and is not required to validate the session token.

**Comment by KarinaBanana** [comment]
Ok so after trying every single easy options for a woman that doesn't understand anything about tokens and servers, i found the solution. I don't really know how, but I'll share with you. 

I went on the tinder app page on facebook and registered the app. Then in my facebook controls I made sure it was registered and activated and blablabla. 
I then deleated my facebook application and in my phone settings, I cheked the applications that has access to my facebook. Tinder was there. I desactivated access and re-activated it, then put my facebook password, download facebook and it worked!!
